internal/parser/reparser.go

```diff
@@ -178,6 +182,39 @@ func (p *Parser) reparseTags(parent *ast.Node, jsDoc []*ast.Node) {
 						fun.FunctionLikeData().Type = p.makeNewType(tag.AsJSDocReturnTag().TypeExpression, fun)
 					}
 				}
+			case ast.KindJSDocReadonlyTag, ast.KindJSDocPrivateTag, ast.KindJSDocPublicTag, ast.KindJSDocProtectedTag, ast.KindJSDocOverrideTag:
+				switch parent.Kind {
+				case ast.KindPropertyDeclaration, ast.KindMethodDeclaration, ast.KindGetAccessor, ast.KindSetAccessor:
+					// !!! BinaryExpression (this.p assignments)
+					var keyword ast.Kind
+					switch tag.Kind {
+					case ast.KindJSDocReadonlyTag:
+						keyword = ast.KindReadonlyKeyword
+					case ast.KindJSDocPrivateTag:
+						keyword = ast.KindPrivateKeyword
+					case ast.KindJSDocPublicTag:
+						keyword = ast.KindPublicKeyword
+					case ast.KindJSDocProtectedTag:
+						keyword = ast.KindProtectedKeyword
+					case ast.KindJSDocOverrideTag:
+						keyword = ast.KindOverrideKeyword
+					}
+					modifier := p.factory.NewModifier(keyword)
+					modifier.Loc = tag.Loc
+					modifier.Flags = p.contextFlags | ast.NodeFlagsReparsed
+					var nodes []*ast.Node
+					var loc core.TextRange
+					if parent.Modifiers() == nil {
+						nodes = p.nodeSlicePool.NewSlice(1)
+						nodes[0] = modifier
+						loc = tag.Loc
+					} else {
+						nodes = append(parent.Modifiers().Nodes, modifier)
+						loc = parent.Modifiers().Loc
+					}
+					parent.AsMutable().SetModifiers(p.newModifierList(loc, nodes))
```

Comment:
Does this ever happen on _any_ parent kind, or is there a specific subset of nodes that do this? I'm not entirely certain how I feel about the `AsMutable` pattern, so curious if it's just shorthand for a couple of types.